sdk/metric/reader.go

```diff
@@ -56,13 +56,17 @@ type Reader interface {
 	// The producer argument allows the Reader to signal the sdk to collect
 	// and send aggregated metric measurements.
 	register(sdkProducer)
+	// DO NOT CHANGE: any modification will not be backwards compatible and
```

Comment:
We do not allow user-defined readers. Therefore should we add a following notice?

> Warning: methods may be added to this interface in minor releases.

Reply:
This is an exported interface with exported methods. Modifying the exported methods in any way is a breaking change once they are released stability regardless of if we allow user defined readers or not. This comment is the conclusion of the community decision in #1714 to ensure we make it clear to future developers to not introduce breaking changes here.

Reply:
Take notice that this is an interface based on specification so a method can be added.

Reply:
Changed.